micromanager_gui/main_window.py

```diff
@@ -267,13 +272,37 @@ def _refresh_camera_options(self):
                 )
 
     def _refresh_objective_options(self):
-        if "Objective" in self._mmc.getLoadedDevices():
-            with blockSignals(self.objective_comboBox):
-                self.objective_comboBox.clear()
-                self.objective_comboBox.addItems(self._mmc.getStateLabels("Objective"))
-                self.objective_comboBox.setCurrentText(
-                    self._mmc.getStateLabel("Objective")
-                )
+        for cfg in self._mmc.getAvailableConfigGroups():
+            if OBJ_PTRN.match(cfg):
+
+                cfg_groups_options = self._mmc.getAvailableConfigs(cfg)
+                cfg_groups_options_keys = (
+                    self._mmc.getConfigData(cfg, cfg_groups_options[0])
+                ).dict()
+
+                dev_name = [
+                    k
+                    for idx, k in enumerate(cfg_groups_options_keys.keys())
+                    if idx == 0
+                ][0]
+
+                self.objectives_device = dev_name
+
+                with blockSignals(self.objective_comboBox):
+                    self.objective_comboBox.clear()
+                    self.objective_comboBox.addItems(
+                        self._mmc.getAvailableConfigs(self.objectives_device)
+                    )
+                    self.set_pixel_size()
+                    return
+
+        for device in self._mmc.getLoadedDevicesOfType(DeviceType.StateDevice):
+            if OBJ_PTRN.match(device):
+                self.objectives_device = device
```

Comment:
Maybe this could actually live in `pymmocre-plus`? Seems generically useful and also thus far pymmcoreplus has been where the complexity and idiosyncracies of pymmcore has been smoothed over.

Reply:
somehting like this on `CMMCorePlus`

```python
import re
from typing import List, Pattern
from pymmcore_plus import DeviceType
def getObjectiveDevices(self, device_regex: Pattern=None)-> List[str]:
    
    devices = []
    if device_regex is None:
        device_regex = re.compile("(.+)(Nosepiece|Objective|obj)s?", re.IGNORECASE)
    
    for device in self.getLoadedDevicesOfType(DeviceType.StateDevice):
        if device_regex.match(device):
            devices.append(device)
    return devices
```

Reply:
See https://github.com/tlambert03/pymmcore-plus/pull/53 for a proposal